app/sessions.go

```diff
@@ -0,0 +1,53 @@
+package app
+
+import (
+	"database/sql"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+type Session struct {
+	UserId    int
+	SessionId int
+	ExpiresAt time.Time
+}
+
+type SessionService struct {
+	db *sql.DB
+}
+
+func NewSessionService(db *sql.DB) *SessionService {
+	return &SessionService{
+		db: db,
+	}
+}
+
+func (s *SessionService) SaveSession(userID int, expiresAt time.Time) (int, error) {
+	var sessionID int
+	err := s.db.QueryRow("INSERT INTO sessions (user_id, expires_at) VALUES ($1, $2) RETURNING session_id", userID, expiresAt).Scan(&sessionID)
```

Comment:
The session table should have it's own primary key `id` which is autoincremented. But the `session_id` should just be a string (no need for it to be a numeric type) that is randomly generated for each session. You can check out online how session IDs should be generated and let me know if you need docs on this. But using the autoincremented integer as session id is not correct, because you have a pattern to creating these and the issue is that you will run out of integers at some point, which you can guess is not a good thing.